src/bsoncxx/test/bson_builder.cpp

```diff
@@ -1246,95 +1247,181 @@ TEST_CASE("builder::stream::document throws on consecutive keys", "[bsoncxx::bui
     bson_t expected;
     bson_init(&expected);
 
     bson_append_utf8(&expected, "hello", -1, "world", -1);
 
     SECTION("works with string literals") {
-        builder::list b{"hello", "world"};
+        builder::list::document b{"hello", "world"};
 
-        bson_eq_object(&expected, b.view().get_document().value);
+        bson_eq_object(&expected, b.extract().view());
     }
 
     SECTION("works with std::string") {
-        builder::list b{"hello", std::string{"world"}};
+        builder::list::document b{"hello", std::string{"world"}};
 
-        bson_eq_object(&expected, b.view().get_document().value);
+        bson_eq_object(&expected, b.extract().view());
     }
 
     SECTION("works with b_string") {
-        builder::list b{"hello", types::b_string{"world"}};
+        builder::list::document b{"hello", types::b_string{"world"}};
 
-        bson_eq_object(&expected, b.view().get_document().value);
+        bson_eq_object(&expected, b.extract().view());
     }
 
     SECTION("works with const char*") {
         const char* world = "world";
-        builder::list b{"hello", world};
+        builder::list::document b{"hello", world};
 
-        bson_eq_object(&expected, b.view().get_document().value);
+        bson_eq_object(&expected, b.extract().view());
     }
 
     SECTION("works with char*") {
         char* world = const_cast<char*>("world");
-        builder::list b{"hello", world};
+        builder::list::document b{"hello", world};
 
-        bson_eq_object(&expected, b.view().get_document().value);
+        bson_eq_object(&expected, b.extract().view());
     }
 
     bson_destroy(&expected);
 }
 
-TEST_CASE("list builder appends double", "[bsoncxx::builder::list]") {
+TEST_CASE("list builder document copy and move semantics", "[bsoncxx::builder::list::document]") {
+    bson_t expected;
+    bson_init(&expected);
+
+    bson_append_utf8(&expected, "hello", -1, "world", -1);
+
+    SECTION("move constructor") {
+        builder::list::document original{"hello", "world"};
+        auto moved{std::move(original)};
+        bson_eq_object(&expected, moved.extract().view());
+    }
+
+    SECTION("move assignment") {
+        builder::list::document original{"hello", "world"};
+        auto moved = std::move(original);
+        bson_eq_object(&expected, moved.extract().view());
+    }
+
+    SECTION("copy constructor") {
+        builder::list::document original{"hello", "world"};
+        builder::list::document copied{original};
+        bson_eq_object(&expected, copied.extract().view());
+    }
+
+    SECTION("copy assignment") {
+        builder::list::document original{"hello", "world"};
+        builder::list::document copied{"wrong", "values"};
+        copied = original;
+
+        bson_eq_object(&expected, copied.extract().view());
+    }
+
+    SECTION("self copy assignment") {
+        builder::list::document actual{"hello", "world"};
+        actual = *&actual;  // '*&' suppresses -Wself-assign-overloaded warning
```

Comment:
ooh, fascinating

Reply:
Removed this since the copy constructors no longer exist.

Reply:
👍